extensions/amp-video-service/0.1/amp-video-service.js

```diff
@@ -76,14 +148,159 @@ export class VideoService {
 
   /**
    * @param {!AmpElement} unusedVideo
-   * @param {!../../../src/observable.Observable<boolean>} unusedObservable
+   * @param {!Observable<boolean>} unusedObservable
    */
   delegateAutoplay(unusedVideo, unusedObservable) {
     warnUnimplemented('Autoplay delegation');
   }
 }
 
 
+/** @visibleForTesting */
+export class VideoEntry {
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  constructor(ampdoc, videoService, video) {
+
+    /** @private @const{!../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const {!VideoService} */
+    this.service_ = videoService;
+
+    /** @private @const{!../../../src/video-interface.VideoInterface} */
+    this.video_ = video;
+
+    /** @visibleForTesting {boolean} */
+    this.isPlaying = false;
+  }
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   * @param {!VideoService} videoService
+   * @param {!../../../src/video-interface.VideoInterface} video
+   */
+  static create(ampdoc, videoService, video) {
+    return new VideoEntry(ampdoc, videoService, video);
+  }
+
+  /** @param {function()} handler */
+  onPlaybackTick(handler) {
+    this.service_.onTick(() => {
+      if (!this.isPlaying) {
+        return;
+      }
+      handler();
+    });
+  }
+
+  /** */
+  install() {
+    const {element} = this.video_;
+    const signals = element.signals();
+
+    element.dispatchCustomEvent(VideoEvents.REGISTERED);
+
+    // Unlike events, signals are permanent. We can wait for `REGISTERED` at any
+    // moment in the element's lifecycle and the promise will resolve
+    // appropriately each time.
+    signals.signal(VideoEvents.REGISTERED);
+
+    element.whenBuilt()
+        .then(() => this.onBuilt_());
+
+    signals.whenSignal(CommonSignals.LOAD_START)
+        .then(() => this.onLoadStart_());
+  }
+
+  /** @private */
+  onBuilt_() {
+    const {element} = this.video_;
+
+    this.registerCommonActions();
+    this.addEventHandlers_();
+
+    element.classList.add('i-amphtml-video-interface');
+  }
+
+  /** @private */
+  onLoadStart_() {
+    this.maybeTriggerTimeUpdate();
+  }
+
+  /** @private */
+  addEventHandlers_() {
+    const {element} = this.video_;
+
+    listen(element, VideoEvents.PAUSE, () => {
+      this.isPlaying = false;
+    });
+
+    listen(element, VideoEvents.PLAYING, () => {
+      this.isPlaying = true;
+    });
+  }
+
+  /**
+   * Register common actions such as play, pause, etc... so they can be called
+   * using AMP Actions, e.g.: `<button on="tap:myVideo.play">`.
+   * @visibleForTesting
+   */
+  registerCommonActions() {
```

Comment:
You can still mark these `@private` and access them normally in unit tests (not integration tests I believe) to allow CC renaming.

Reply:
Done.